### PR TITLE
Catalog platform-a HBA and Aggregator information

### PR DESCRIPTION
**Pull request by jlongever**
- Collect and catalog Platform-A HBA and Aggregrator resource information; adds the data to the Chassis (enclosure) node.

@RackHD/corecommitters @uppalk1 @tannoa2 @zyoung51 @keedya

### CODE REVIEW

**Comment by jlongever** [comment]
test this please

File: lib/jobs/emc-redfish-catalog.js [hunk 1]
```diff
@@ -56,27 +58,155 @@ function EmcRedfishCatalogJobFactory(
      */
     EmcRedfishCatalogJob.prototype._run = function() {
         var self = this;
-        return self.catalogElements()
+        return Promise.resolve(self.chassis)
+        .each(self.catalogElements.bind(self))
```
Comment (benbp) [inline lib/jobs/emc-redfish-catalog.js]: Can these be parallelized?

**Comment by benbp** [comment]
👍

**Comment by yyscamper** [comment]
:+1:

File: lib/jobs/emc-redfish-catalog.js (hunk 1 above)
Comment (jlongever) [inline lib/jobs/emc-redfish-catalog.js]: @benbp Probably not too much difference as-is, typically the chassis list size is 1 but I could see a benefit as the chassis/catalog grows.

**Comment by jlongever** [comment]
Merging, made note of potential future optimizations.

**Pull request merged**